src/browser/components/MainPage.jsx

```diff
@@ -283,6 +294,19 @@ export default class MainPage extends React.Component {
     }
   }
 
+  handleDoubleClick = () => {
+    // TODO Devin: Implement for Windows
+    const doubleClickAction = remote.systemPreferences.getUserDefault('AppleActionOnDoubleClick', 'string');
+    const win = remote.getCurrentWindow();
+    if (doubleClickAction === 'Minimize') {
+      win.minimize();
+    } else if (doubleClickAction === 'Maximize' && !win.isMaximized()) {
+      win.maximize();
+    } else if (doubleClickAction === 'Maximize' && win.isMaximized()) {
+      win.unmaximize();
+    }
```

Comment:
Maybe you want to have here some sort of strategy pattern, where you don't really care if it's win or mac, but define the behaviours on initialization and here you just apply whatever is configured.

Reply:
So there's two completely different systems at play here. The Mac one relies on this function and a `drag` region, whereas the Windows/Linux one is a custom toolbar npm package that handles all this. I'm not really sure how I'd want to 'inject' them separately, since the app doesn't really provides the correct points of insertion, nor do they require the same points.

I think if we were to change the application to be more dependency injection receptive, then it might make sense to modularize these.